### PR TITLE
fix: improved error message when eventual send target is undefined

### LINKED ISSUE #1701: zcf.saveIssuer() with bad issuer - improve error message
**What is the Problem Being Solved?**

If we call `zcf.saveIssuer()` with a bad issuer, we get the following unhelpful error message:

```
    message: 'o["getBrand"] is not a function',
```

**Description of the Design**

Improve this error message by either doing an assertion before this call (an extra trip) or catching any errors related to the issuer not having the expected API and inserting a custom, improved error message.

**Security Considerations**

If we catch errors, we should make sure that we are not suppressing errors unrelated to the issuer not having the expected API.

**Test Plan**
The ZCF tests that found the issue should confirm the error message has improved. We may want to have additional tests to ensure we aren't suppressing errors unrelated.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
fixes: #1701

### CODE REVIEW

**Review by michaelfig — CHANGES_REQUESTED** [review]
Good start!

Can you adjust the other errors in this function to have a similar style?

File: packages/eventual-send/src/index.js [hunk 1]
```diff
@@ -361,27 +361,26 @@ export function makeHandledPromise() {
       if (method === undefined || method === null) {
         if (!(t instanceof Function)) {
           const ftype = typeof t;
-          throw TypeError(`target is not a function, typeof is ${ftype}`);
+          throw TypeError(
+            `method name was undefined. (target's type is "${ftype}")`,
```
Comment (Chris-Hibbert) [inline packages/eventual-send/src/index.js]: I tested this manually in the debugger, by changing a message to `undefined` just before this point. If you can think of a way to add a test that sends `undefined`, I'd add it to the tests.

**Comment by Chris-Hibbert** [comment]
> Can you adjust the other errors in this function to have a similar style?

Done. 

What do you think of these? I couldn't find javascript standard terminology for receiver/target, so I used words that sounded natural to me.

**Comment by Chris-Hibbert** [comment]
@michaelfig, having improved a few of the error messages, I find that SwingSet's `test-network.js` now fails because it hits one of the new messages. Except I didn't change the logic of when the Error is thrown, just the contained message. Is something catching (or otherwise handling) the old Error by recognizing the string? I can't find anyplace that is looking for the old message, and it appears that the error it detects is real.

```
network › loopback protocol

  Rejected promise returned by test. Reason:

  TypeError {
    message: 'target has no method "onListen", has [onAccept]',
  }
```

**Comment by michaelfig** [comment]
> Is something catching (or otherwise handling) the old Error by recognizing the string? I can't find anyplace that is looking for the old message, and it appears that the error it detects is real.

Could you take a look at my ugly hack in https://github.com/Agoric/agoric-sdk/blob/32253baa747f7111b66846709465a5892fd84f65/packages/SwingSet/src/vats/network/network.js#L22

**Comment by Chris-Hibbert** [comment]
> Could you take a look at ...

Thanks! Now what do you think of the new messages?

@katelynsills, I'm adding you as a reviewer because I changed the messages identified in issue #1701. I think it's clearer now, but it still doesn't identify the problem as a being caused by passing a non-Issuer.

**Review by katelynsills — APPROVED** [review]
I just looked at the zcf error change, but that is much, much better. Thanks for this change!

File: packages/eventual-send/src/index.js (hunk 1 above)
Comment (michaelfig) [inline packages/eventual-send/src/index.js]:
```suggestion
            `Cannot invoke target as a function, it is a ${ftype}`,
```

Comment (michaelfig) [reply packages/eventual-send/src/index.js]: > If you can think of a way to add a test that sends `undefined`, I'd add it to the tests.

`E(target)(...args)` sends `undefined`, as it is a function invocation, not a method call (though not supported by our current marshal layer).

You should be able to get it to work/fail somewhere in the eventual-send tests.

**Review by michaelfig — APPROVED** [review]
It's looking really good!  Once you add the `E(target)(...args)` test, feel free to merge.

**Pull request merged**